modules/runner/src/main/java/org/apache/ignite/runner/internal/app/IgnitionImpl.java

```diff
@@ -0,0 +1,179 @@
+
+        List<ConfigurationStorage> configurationStorages =
+            new ArrayList<>(Collections.singletonList(new LocalConfigurationStorage(vaultMgr)));
+
+        // Bootstrap local configuration manager.
+        ConfigurationManager locConfigurationMgr = new ConfigurationManager(rootKeys, configurationStorages);
+
+        if (!cfgBootstrappedFromPds)
+            try {
+                locConfigurationMgr.bootstrap(jsonStrBootstrapCfg);
+            }
+            catch (Exception e) {
+                log.warn("Unable to parse user specific configuration, default configuration will be used", e);
+            }
+        else if (jsonStrBootstrapCfg != null)
+            log.warn("User specific configuration will be ignored, cause vault was bootstrapped with pds configuration");
+
+        NetworkView netConfigurationView =
+            locConfigurationMgr.configurationRegistry().getConfiguration(NetworkConfiguration.KEY).value();
+
+        var serializationRegistry = new MessageSerializationRegistry();
+
+        // Network startup.
+        ClusterService clusterNetSvc = new ScaleCubeClusterServiceFactory().createClusterService(
+            new ClusterLocalConfiguration(
+                "Node" + netConfigurationView.port(),
+                netConfigurationView.port(),
+                Arrays.asList(netConfigurationView.netMembersNames()),
+                serializationRegistry
+            )
+        );
+
+        // Raft Component startup.
+        Loza raftMgr = new Loza(clusterNetSvc);
+
+        // MetaStorage Component startup.
+        MetaStorageManager metaStorageMgr = new MetaStorageManager(
+            vaultMgr,
+            clusterNetSvc,
+            raftMgr
+        );
+
+        // TODO IGNITE-14578 Bootstrap configuation manager with distributed configuration.
+        configurationStorages.add(new DistributedConfigurationStorage(metaStorageMgr));
+
+        // Start configuration manager.
+        ConfigurationManager configurationMgr = new ConfigurationManager(rootKeys, configurationStorages);
+
+        // Baseline manager startup.
+        BaselineManager baselineMgr = new BaselineManager(configurationMgr, metaStorageMgr, clusterNetSvc);
+
+        // Affinity manager startup.
+        new AffinityManager(configurationMgr, metaStorageMgr, baselineMgr);
+
+        SchemaManager schemaMgr = new SchemaManager(configurationMgr);
+
+        // Distributed table manager startup.
+        TableManager distributedTblMgr = new TableManagerImpl(
+            configurationMgr,
+            clusterNetSvc,
+            metaStorageMgr,
+            schemaMgr,
+            raftMgr
+        );
+
+        // TODO IGNITE-14579 Start rest manager.
+
+        // Deploy all resisted watches cause all components are ready and have registered their listeners.
+        metaStorageMgr.deployWatches();
+
+        clusterNetSvc.start();
```

Comment:
This looks suspicious to me. Having the network started after all components means the metastore is not working before this point and all dependant services will not be able to read any distributed property. 
Do I understand correctly what only local properties are available for services in the constructors ?
At which moment exactly metastore's raft group is started ? This can be done after this call, but I don't see any related callback.
Another case - metastore group is not yet created.
This RP doesn't contain any related tests, which is bad. Can you add unit tests for these scenarios using mocks when needed:
1) New cluster startup.
2) Existing cluster startup.

Reply:
metastore's raft group is started as a result of handling cluster init message that is processed inside `MetaStorageManager#MetaStorageManager`

```        
clusterNetSvc.messagingService().addMessageHandler((message, sender, correlationId) -> {
    // TODO: IGNITE-14414 Cluster initialization flow.
});
```
So before cluster init node is started in "Zombie" state, all managers are ready, network and other components are configured with local properties either with vault's PDS or user-specific bootstrap configuration. Nodes in cluster know about each other because network layer is up and running. All managers registered their intention on watching distributed properties using `MetaStorageManager#registerWatch`. Node is waiting for cluster init message in order to find out whether it should deploy metastorage rafr server.